roles/set_scale/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+# See meta/main.yml for a list of dependency roles that will be executed prior to this role
+---
+- name: Ensure parameters have been set
+  fail: msg="Bailing out. This role requires 'machineset_instance_type' and 'scale' to be defined"
+  when: not machineset_instance_type or not scale
+
+- name: Get current replicas of machinesets with type {{ machineset_instance_type }}
+  command:
+    oc get machineset -n openshift-machine-api
+    -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.spec.replicas}{"\n"}{end}'
+  register: machineset_replicas
+
+- name: Calculate sum of replicas of machinesets with type {{ machineset_instance_type }}
+  set_fact:
+    current_replicas_sum: "{{ machineset_replicas.stdout_lines | map('int') | sum(start=0) }}"
+
+- name: Log sum result
+  debug:
+    msg: "Current sum of replicas is {{ current_replicas_sum }}, wanted {{ scale }}"
+
+- when: current_replicas_sum != scale
+  name: Change all {{ machineset_instance_type }} machinesets replicas to have sum {{ scale }}
+  block:
+  - name: Get the names of existing worker machinesets with type {{ machineset_instance_type }}
+    command:
+      oc get machineset -n openshift-machine-api -o=jsonpath='{range .items[?(@.spec.template.spec.providerSpec.value.instanceType=="{{ machineset_instance_type }}")]}{.metadata.name}{"\n"}{end}'
+    register: oc_get_machinesets
+    failed_when: not oc_get_machinesets.stdout
+  - name: Set the replicas of all {{ machineset_instance_type }} machinesets to 0 except the first
+    command: |
+      oc patch machineset -n openshift-machine-api {{ item }} --patch '{"spec": {"replicas": 0 }}' --type merge
+    loop: "{{ oc_get_machinesets.stdout_lines[1:] }}"
+  - name: Get first machineset
+    set_fact:
+      first_machineset: "{{ oc_get_machinesets.stdout_lines[0] }}"
+  - name: Set the replicas of the first {{ machineset_instance_type }} machineset ({{ first_machineset }}) to {{ scale }}
+    command: >
+      oc patch machineset -n openshift-machine-api {{ first_machineset }}
+      --patch '{"spec": {"replicas": {{ scale }} }}' --type merge
```

Comment:
I'm not sure modifying *all* the machinesets [with the given machine-type] is the best strategy,
maybe you could change it to something like:

is there only one MachineSet generating type `machine-type`? yes, modify it to `scale`
no --> cannot perform task ==> allow receiving a machineset name change change only this replica to `scale`

Reply:
I thought the idea was to make sure the number of machines of given instance type, across the entire cluster, is exactly `scale`.

If we ask the user to provide us with a machineset name that needs to be modified - then the role doesn't really do its job, it only modifies that specific machineset, and doesn't achieve the goal of setting the cluster-wide total to exactly `scale`.

Here's an alternative suggestion - if there are other machinesets in the cluster with this instance type that need to be downscaled to 0 in order to achieve the required cluster-wide `scale`, then the role will ask the user to provide a --force flag telling them it's going to happen, so we don't perform anything extreme without explicit confirmation from the user.

Reply:
> I thought the idea was to make sure the number of machines of given instance type, across the entire cluster, is exactly `scale`.

The main (and only) requirement is to allow the CI to scale up and down the # of GPU nodes,
so in this environment, we know that only one MachineSet of the given instance-type will be available.
The question could be solved by aborting the play if multiple MachineSets are found.

But I like the `--force` option, it makes sense to me: default behavior is to do things as simple as possible, force behavior makes sure we have the right count, possibility by scaling down some machinesets: :+1: